env_support/cmake/custom.cmake

```diff
@@ -101,7 +118,7 @@ install(
   PATTERN "*.h")
 
 # install example headers
-if(NOT LV_CONF_BUILD_DISABLE_EXAMPLES)
+if(CONFIG_LV_BUILD_EXAMPLES)
```

Comment:
For all of these, can we do something like `if(NOT DEFINED CONFIG_LV_BUILD_EXAMPLES OR CONFIG_LV_BUILD_EXAMPLES)`? The reason I ask is because users using a CMakeLists.txt like [this](https://gist.github.com/liamHowatt/2b7b101cbcc93ac3a6d687e82f63f816) will have their build break until they add `set(CONFIG_LV_BUILD_EXAMPLES ON)`.

I suppose we are already breaking users of `*_DISABLE_*`

Reply:
this is a design question ;) @kisvegabor  what do you think :)

Reply:
I think enabling demos and examples if `CONFIG_LV_BUILD_EXAMPLES` is not defined is a good idea. We need to think about novice users and a good get started experience.   

Early in the main [`CMakeList.txt`](https://github.com/lvgl/lvgl/blob/master/CMakeLists.txt) file we can define these with a default value (if not defined).

Reply:
I added a FATAL_ERROR if CONFIG_LV_BUILD_EXAMPLES is set but not CONFIG_LV_BUILD_DEMOS.

I'd like to keep all settings in a single place. 
In this is the CMake Presets.json (or alternatively within the cmake invocation `cmake -G ... -DFOO`) is the only place where config values are set.

If we start setting config values implicitly (in this case the root CmakeLists.txt) thing will get much more complicated .
I hope this is ok for you guys

Reply:
ASAN is complaining about a memory leak because i needed to build the examples whithin the test as well.
Does anybody know how to solve this?

Reply:
I'm sorry, I've missed this comment when approving. 

I'm open to where the config options are set, but we shall provide a smooth getting started experience so novice users too. It means `cmake ..; make` should be able to compile the demos and example without reading the docs for extra option. Also note that 

Therefore even if it's more complicated on our side I would prefer adding something like this for demos and examples:
```
if(NOT DEFINED CONFIG_LV_BUILD_EXAMPLES)
    set(CONFIG_LV_BUILD_EXAMPLES ON)
endif()

if(NOT DEFINED CONFIG_LV_BUILD_DEMOS)
    set(CONFIG_LV_BUILD_DEMOS ON)
endif()
```

It's okay to enable ThorVG explicitly as it's not certain that it can be compiled on all systems. And it's considered as an advanced feature.

I've also noticed that the name configs options are structured differenty:
- **CONFIG_LV**_BUILD_EXAMPLES ON
- **CONFIG_LV**_BUILD_DEMOS ON
- **LV_CONF**_BUILD_THORVG_INTERNAL
```

Probably both the Kconfig names  (e.g. `CONFIG_LV_BUILD_EXAMPLES` for compatibility with ESP and Zephyr) and `lv_conf.h` names `LV_BUILD_EXAMPLES` should work. It can result in a config conflict if the examples are disabled in `lv_conf.h` but enabled in `cmake`, however we cannot do much about it. If you start setting the values via `cmake` you should remove them from `lv_conf.h`.

@liamHowatt before merging we should test if the examples can be enabled/disabled e.g. with ESP with these changes.